ONSdigital/eq-questionnaire-runner
# Spike: Deploy Cloud Run using Terraform

### What is the context of this PR?
This spike demonstrates the Cloud Run deployment using Terraform instead of gcloud.

### How to review 
Review using:
- https://github.com/ONSdigital/eq-pipelines/pull/263

OR
- Deploy directly using Terraform:
   - Ensure you have tf 1.0.0 install locally.
   - Run `cp ci/terraform/terraform.tfvars.example ci/terraform/terraform.tfvars` and update the variables in this file if desired.
   - If you use an existing staging env, make sure to delete the existing runner Cloud Run deployment first!. Otherwise, deploy the main infrastructure to a new project using https://github.com/ONSdigital/eq-terraform-gcp
   - Once the infrastructure is up, deploy the app using:
  	 ```bash
  	 PROJECT_ID=<project_id> \
     TERRAFORM_STATE_BUCKET=<state-bucket-name> \
     ./ci/terraform/scripts/deploy.sh
	 ```
   - Destroy the app using:
  	 ```bash
  	 PROJECT_ID=<project_id> \
     TERRAFORM_STATE_BUCKET=<state-bucket-name> \
     ./ci/terraform/scripts/destroy.sh
	 ```

TO DO when implementing:
- Enforce type hints and add validation for the TF vars

### Checklist
